### DISCUSSION TITLE
`python.analysis.diagnosticMode`

### DISCUSSION THREAD

**Discussion by InSyncWithFoo** [microsoft/pyright#7736] [General]
This setting has two possible values: `workspace` and `openFilesOnly`. I get it that this changes the number of diagnostics reported in one `publishDiagnostic` notification, but is that all? Are there other differences that would make one more or less preferable?

In other words, what are their intended usages?

**Comment by erictraut** [comment]
The `workspace` setting typically requires more computation (significantly more in some cases) because any modification to one file will cause pyright to reanalyze any dependent files. If you are modifying a file that many other files in your project depend upon, this could cause hundreds or thousands of files to be reanalyzed after a single keystroke.

That said, I personally prefer to use `workspace` in most cases because I want to understand the impact of any changes I make in my code. If I don't introduce any new typing violations anywhere within the workspace, there's a good chance that I haven't introduced any new bugs when making a change.

Some developers prefer to use pre-commit hooks to run a type checker once prior to committing a set of changes. In that case, it may be preferable to use `openFilesOnly`.
